### DISCUSSION TITLE
Are there any cases where a 'deadlock detected (SQLSTATE 40P01)' error occurs?

### DISCUSSION THREAD

**Discussion by hyeyoung-leee** [goharbor/harbor#20379] [Q&A] [answered]
Hi,

~~Can a deadlock error occur when simultaneously pushing multiple images to a project with storage quota?
`unknown: ERROR: deadlock detected (SQLSTATE 40P01)`~~

I have found the reason.
I want to limit the number of repositories per project, so I have added the code to the file.
`/src/server/middleware/blob/put_manifest.go`

<img width="1563" alt="스크린샷 2024-04-30 오후 4 05 48" src="https://github.com/goharbor/harbor/assets/120620071/20bd1144-20bc-42e8-83eb-98d924634a74">

This sometimes causes a deadlock between `project_blob` and `quota_usage` tables.
Can you help me fix this?

**Comment by MinerYang** [comment, marked as answer]
We do not support add repository limitations for projects, BTW, this is no relevant to either put_manifest or quota refresh.
